### PR TITLE
- Parser::AST::Processor: do not add nil to the end of required argument.

### LINKED ISSUE #298: Parser::AST::Processor modifies sexp
``` ruby
require 'ast'
require 'parser/ruby23'

sexp = Parser::Ruby23.parse("def m(a); end")
p sexp

processed = Parser::AST::Processor.new.process(sexp)
p processed
```
Output:
``` ruby
s(:def, :m,
  s(:args,
    s(:arg, :a)), nil)
s(:def, :m,
  s(:args,
    s(:arg, :a, nil)), nil)
```

`Processor` adds an extra `nil` to the end of any required argument. Should it be fixed? Wondering because it requires some breaking changes in `processor`

### PR DESCRIPTION
**Pull request by iliabylich**
Closes #298

### CODE REVIEW

**Comment by whitequark** [comment]
No, wait, this is not right. `on_argument` is supposed to be overridden, so you need to call it from both `on_optional_argument` and `on_required_argument`.

**Comment by iliabylich** [comment]
But `on_required_argument` should be different from `on_optional_argument`. It should simply return `node`. Or am I missing something?

**Comment by whitequark** [comment]
Maybe just alter `on_argument` to look at the number of children then...

**Comment by iliabylich** [comment]
Optional arg nodes always have two children, right? Then I guess we can just check for `value_node` to be present. Updated PR.

File: lib/parser/ast/processor.rb [hunk 1]
```diff
@@ -39,9 +39,13 @@ def process_variable_node(node)
       def on_vasgn(node)
         name, value_node = *node
 
-        node.updated(nil, [
-          name, process(value_node)
-        ])
+        if value_node
```
Comment (whitequark) [inline lib/parser/ast/processor.rb]: `if !value_node.nil?`

Comment (IlyaBylich) [inline lib/parser/ast/processor.rb]: It can be false? Could you provide an example of such case? I can't get it locally

Comment (whitequark) [inline lib/parser/ast/processor.rb]: It can't. The code I suggest conveys intent more clearly.

**Comment by iliabylich** [comment]
@whitequark Fixed

**Pull request merged**